src/components/ha-target-picker.ts

```diff
@@ -251,10 +251,8 @@ export class HaTargetPicker extends SubscribeMixin(LitElement) {
   private async _showPicker(ev) {
     this._addMode = ev.currentTarget.type;
     await this.updateComplete;
-    setTimeout(() => {
-      this._inputElement?.open();
-      this._inputElement?.focus();
-    }, 0);
+    await this._inputElement?.focus();
+    this._inputElement?.open();
```

Comment:
```suggestion
    await this._inputElement?.open();
```

Just for clarity and safety later on when a parent might want to show it.